Source/ARTAuth.m

```diff
@@ -271,11 +272,11 @@ - (NSMutableURLRequest *)buildRequest:(ARTAuthOptions *)options withParams:(ARTT
     if ([options isMethodPOST]) {
         // TokenParams take precedence over any configured authParams when a name conflict occurs
         NSDictionary *unitedParams = [params toDictionaryWithUnion:options.authParams];
```

Comment:
It seems that `toDictionaryWithUnion:` returns `<NSString *, NSString *> *` so could you consider:
1. Using that specific type for `unitedParams`
2. Replacing use of `description` on `id` pointers elsewhere (e.g. `enc` function) with `NSString *`